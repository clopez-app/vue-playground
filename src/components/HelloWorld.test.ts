import HelloWorld from './HelloWorld.vue';
import { mount } from '@vue/test-utils';

describe('Hello world component', () => {
  test('renders correctly', () => {
    expect(HelloWorld).toBeTruthy();

    const wrapper = mount(HelloWorld, {
      props: {
        msg: 'Test message',
      },
    });

    expect(wrapper.text()).toContain('Test message');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
