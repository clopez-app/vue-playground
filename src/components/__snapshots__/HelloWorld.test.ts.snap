// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hello world component > renders correctly 1`] = `
"<div data-v-e17ea971=\\"\\" class=\\"hello\\">
  <h1 data-v-e17ea971=\\"\\">Test message</h1>
  <p data-v-e17ea971=\\"\\"> For a guide and recipes on how to configure / customize this project,<br data-v-e17ea971=\\"\\"> check out the <a href=\\"https://cli.vuejs.org\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">vue-cli documentation</a>. </p>
  <h3 data-v-e17ea971=\\"\\">Installed CLI Plugins</h3>
  <ul data-v-e17ea971=\\"\\">
    <li data-v-e17ea971=\\"\\"><a href=\\"https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-typescript\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">typescript</a></li>
  </ul>
  <h3 data-v-e17ea971=\\"\\">Essential Links</h3>
  <ul data-v-e17ea971=\\"\\">
    <li data-v-e17ea971=\\"\\"><a href=\\"https://vuejs.org\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">Core Docs</a></li>
    <li data-v-e17ea971=\\"\\"><a href=\\"https://forum.vuejs.org\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">Forum</a></li>
    <li data-v-e17ea971=\\"\\"><a href=\\"https://chat.vuejs.org\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">Community Chat</a></li>
    <li data-v-e17ea971=\\"\\"><a href=\\"https://twitter.com/vuejs\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">Twitter</a></li>
    <li data-v-e17ea971=\\"\\"><a href=\\"https://news.vuejs.org\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">News</a></li>
  </ul>
  <h3 data-v-e17ea971=\\"\\">Ecosystem</h3>
  <ul data-v-e17ea971=\\"\\">
    <li data-v-e17ea971=\\"\\"><a href=\\"https://router.vuejs.org\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">vue-router</a></li>
    <li data-v-e17ea971=\\"\\"><a href=\\"https://vuex.vuejs.org\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">vuex</a></li>
    <li data-v-e17ea971=\\"\\"><a href=\\"https://github.com/vuejs/vue-devtools#vue-devtools\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">vue-devtools</a></li>
    <li data-v-e17ea971=\\"\\"><a href=\\"https://vue-loader.vuejs.org\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">vue-loader</a></li>
    <li data-v-e17ea971=\\"\\"><a href=\\"https://github.com/vuejs/awesome-vue\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">awesome-vue</a></li>
  </ul>
</div>"
`;
